src/full/Agda/Interaction/Options/Base.hs

```diff
@@ -1123,109 +1448,204 @@ deadStandardOptions =
+--
+pragmaFlag :: (IsBool a, KnownBool b)
+  => String
+       -- ^ Long option name.  Prepended with @no-@ for negative version.
+  -> Lens' (WithDefault' a b) PragmaOptions
+       -- ^ Field to switch.
+  -> String
+       -- ^ Explanation for positive option.
+  -> String
+       -- ^ Additional info for positive option (not repeated for negative option).
+  -> Maybe String
+       -- ^ Explanation for negative option.
+  -> [OptDescr (Flag PragmaOptions)]
+pragmaFlag long field = pragmaFlag' long field (const return)
+
+-- | Construct a flag of type @WithDefault _@
+--
+pragmaFlag' :: (IsBool a, KnownBool b)
+  => String
+       -- ^ Long option name.  Prepended with @no-@ for negative version.
+  -> Lens' (WithDefault' a b) PragmaOptions
+       -- ^ Field to switch.
+  -> (a -> Flag PragmaOptions)
+       -- ^ Given the new value, perform additional effect (can override field setting).
+  -> String
+       -- ^ Explanation for positive option.
+  -> String
+       -- ^ Additional info for positive option (not repeated for negative option).
+  -> Maybe String
+       -- ^ Explanation for negative option.
+  -> [OptDescr (Flag PragmaOptions)]
+       -- ^ Pair of option descriptors (positive, negative)
+pragmaFlag' long field = pragmaFlagBool' long (field . lensCollapseDefault)
+
+-- | Construct a flag of type 'IsBool'.
+--
+pragmaFlagBool :: (IsBool a)
+  => String
+       -- ^ Long option name.  Prepended with @no-@ for negative version.
+  -> Lens' a PragmaOptions
+       -- ^ Field to switch.
+  -> String
+       -- ^ Explanation for positive option.
+  -> String
+       -- ^ Additional info for positive option (not repeated for negative option).
+  -> Maybe String
+       -- ^ Explanation for negative option.
+  -> [OptDescr (Flag PragmaOptions)]
+pragmaFlagBool long field = pragmaFlagBool' long field (const return)
+
+-- | Construct a flag of type 'IsBool' with extra effect.
+--
+pragmaFlagBool' :: IsBool a
+  => String
+       -- ^ Long option name.  Prepended with @no-@ for negative version.
+  -> Lens' a PragmaOptions
+       -- ^ Field to switch.
+  -> (a -> Flag PragmaOptions)
+       -- ^ Given the new value, perform additional effect (can override field setting).
+  -> String
+       -- ^ Explanation for positive option.
+  -> String
+       -- ^ Additional info for positive option (not repeated for negative option).
+  -> Maybe String
+       -- ^ Explanation for negative option.
+  -> [OptDescr (Flag PragmaOptions)]
+       -- ^ Pair of option descriptors (positive, negative)
+pragmaFlagBool' long field effect pos info neg =
+  [ Option [] [no b long] (flag b) (def b $ expl b) | b <- [True,False] ]
+  where
+  b0     = defaultPragmaOptions ^. field
+  no   b = applyUnless b ("no-" ++)
+  flag b = NoArg $ effect a . set field a
+    where a = fromBool b
+  def  b = applyWhen (fromBool b == b0) (++ " (default)")
+  expl b = if b then unwords [pos, info] else fromMaybe ("do not " ++ pos) neg
```

Comment:
Fixed this `unwords` (case `null info`) by https://github.com/agda/agda/commit/bd4f880a88fa32d301b45c31d86b8547a2b229ed